aws/aws.go

```diff
@@ -69,3 +78,24 @@ func WriteToShell(c *Credentials, windows bool, w io.Writer) {
 		)
 	}
 }
+
+// GetNonExpiredCredentials Return profiles which have a aws_expiration key but are not yet expired
+func GetNonExpiredCredentials(filename string) (Profiles, error) {
```

Comment:
I would return a *pointer* to `Profiles`, that is:

`func GetNonExpiredCredentials(filename string) (*Profiles, error) {}`

And then return `nil` on error. It's more common in Golang to do it this way, and it makes it clearer there was a problem since we return `nil` and not an empty slice.